### ISSUE TITLE
add `require-brand-type-parameter` rule

### ISSUE THREAD

**Issue by marcalexiei** [marcalexiei/eslint-plugin-zod-x#87]
### 🧩 Description

Enforce that all `.brand()` calls always include a type parameter

### ⚙️ Options

None

### ✅ Valid Code Examples

```typescript
z.string().brand<'foo'>()
```

### ❌ Invalid Code Examples

```typescript
z.string().brand()
```

### ✒️ Additional context

- https://zod.dev/api#branded-types
